### PR TITLE
fix(dashboards): HogQL support for dashboard filters on insights with property group filters

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

HogQL `apply_dashboard_filters` only worked if the insight's query uses filters in list form – but typically these days, property filter groups are used. This failed ([Sentry](https://posthog.sentry.io/issues/5260534837/?project=-1&query=is%3Aunresolved&referrer=issue-stream&sort=trends&statsPeriod=7d&stream_index=4&utc=true)).

## Changes

Fixed the bug.

## How did you test this code?

Updated `test_insight_refreshing_query` to use a property filter group instead of just list.

### CODE REVIEW

File: posthog/hogql_queries/query_runner.py [hunk 1]
```diff
@@ -429,15 +432,26 @@ def apply_dashboard_filters(self, dashboard_filter: DashboardFilter) -> Q:
...
+                                if isinstance(self.query.properties, list)
+                                else PropertyGroupFilterValue(**self.query.properties.model_dump()),
+                                PropertyGroupFilterValue(
+                                    type=FilterLogicalOperator.AND, values=dashboard_filter.properties
+                                ),
+                            ],
+                        )
+                    except Exception as e:
+                        # If pydantic is unhappy about the shape of data, let's ignore property filters and carry on
+                        capture_exception()
```
Comment (webjunkie) [inline posthog/hogql_queries/query_runner.py:449]: Curious: Do we have Sentry configured to capture exception-level log message as exceptions? If so, we don't need to capture separately, as below is sufficient.

File: posthog/hogql_queries/query_runner.py [hunk 2]
```diff
@@ -429,15 +432,26 @@ def apply_dashboard_filters(self, dashboard_filter: DashboardFilter) -> Q:
...
+                        logger.error(
+                            "Failed to apply dashboard property filters",
+                            exception=e,
+                            exc_info=True,
```
Comment (webjunkie) [inline posthog/hogql_queries/query_runner.py:453]: Can also use `logger.exception` and then you don't need the `exc_info=True`

**Review by webjunkie — APPROVED** [review]

File: posthog/hogql_queries/query_runner.py (hunk 1 above)
Comment (Twixes) [reply posthog/hogql_queries/query_runner.py:449]: We don't, but there's a way: https://github.com/kiwicom/structlog-sentry

File: posthog/hogql_queries/query_runner.py (hunk 2 above)
Comment (Twixes) [reply posthog/hogql_queries/query_runner.py]: Oh yeah, that's exactly what I was looking for

**Pull request merged**